env.c

```diff
@@ -178,6 +186,19 @@ mkrule(char *name)
 	return r;
 }
 
+static void
+delrule(void *p)
+{
+	struct rule *r = p;
+	
+	if(!r)
+		return;
```

Comment:
Is it ever the case that we have a NULL rule here? I'd prefer to remove this check if it is unneeded.

Reply:
I haven't read the whole code base, so I don't for sure. probably unneeded, so I'll remove both this check and the rule-name check, since `mkrule` allocates a name.

Reply:
The rule name check should not be needed either.

I believe the same should be true for pools as well.

Reply:
just edited the pools. was experimenting with git rebase, and it merged the amended commit smoothly.

Reply:
Also after I had split the comment into 2 and pushed the first, I had lost my comments, but was able to find them with `git reflog`. That's next level!

Reply:
Thanks. Yeah, `git rebase` and `git reflog` are very handy tools, indeed.